# HG changeset patch
# User Rik <rik@octave.org>
# Date 1424839859 28800
#      Tue Feb 24 20:50:59 2015 -0800
# Node ID 0a3ca546d7fc621b3904601baff9ccbddfae859c
# Parent  6645488466c764e581995b5bc9599767148fca46
doc: Add qmr to documentation.

* linalg.txi: Add DOCSTRING entry for qmr in specialized solvers.

bicg.m, bicgstab.m, cgs.m, gmres.m, qmr.m: Add qmr to seealso links.

diff --git a/doc/interpreter/linalg.txi b/doc/interpreter/linalg.txi
--- a/doc/interpreter/linalg.txi
+++ b/doc/interpreter/linalg.txi
@@ -212,8 +212,10 @@ detect a non-Hermitian matrix.
 @DOCSTRING(bicg)
 
 @DOCSTRING(bicgstab)
 
 @DOCSTRING(cgs)
 
 @DOCSTRING(gmres)
 
+@DOCSTRING(qmr)
+
diff --git a/scripts/sparse/bicg.m b/scripts/sparse/bicg.m
--- a/scripts/sparse/bicg.m
+++ b/scripts/sparse/bicg.m
@@ -62,17 +62,17 @@
 ## @item @var{relres} is the final value of the relative residual.
 ##
 ## @item @var{iter} is the number of iterations performed.
 ##
 ## @item @var{resvec} is a vector containing the relative residual at each
 ## iteration.
 ## @end itemize
 ##
-## @seealso{bicgstab, cgs, gmres, pcg}
+## @seealso{bicgstab, cgs, gmres, pcg, qmr}
 ##
 ## @end deftypefn
 
 ## Author: Sylvain Pelissier <sylvain.pelissier@gmail.com>
 ## Author: Carlo de Falco
 
 function [x, flag, res1, k, resvec] = bicg (A, b, tol, maxit, M1, M2, x0)
 
diff --git a/scripts/sparse/bicgstab.m b/scripts/sparse/bicgstab.m
--- a/scripts/sparse/bicgstab.m
+++ b/scripts/sparse/bicgstab.m
@@ -62,17 +62,17 @@
 ## @item @var{relres} is the final value of the relative residual.
 ##
 ## @item @var{iter} is the number of iterations performed.
 ##
 ## @item @var{resvec} is a vector containing the relative residual at each
 ## iteration.
 ## @end itemize
 ##
-## @seealso{bicg, cgs, gmres, pcg}
+## @seealso{bicg, cgs, gmres, pcg, qmr}
 ##
 ## @end deftypefn
 
 function [x, flag, relres, iter, resvec] = bicgstab (A, b, tol, maxit,
                                                      M1, M2, x0)
 
   if (nargin >= 2 && nargin <= 7 && isvector (full (b)))
 
diff --git a/scripts/sparse/cgs.m b/scripts/sparse/cgs.m
--- a/scripts/sparse/cgs.m
+++ b/scripts/sparse/cgs.m
@@ -62,17 +62,17 @@
 ## @item @var{relres} is the final value of the relative residual.
 ##
 ## @item @var{iter} is the number of iterations performed.
 ##
 ## @item @var{resvec} is a vector containing the relative residual at
 ## each iteration.
 ## @end itemize
 ##
-## @seealso{pcg, bicgstab, bicg, gmres}
+## @seealso{pcg, bicgstab, bicg, gmres, qmr}
 ## @end deftypefn
 
 function [x, flag, relres, iter, resvec] = cgs (A, b, tol, maxit, M1, M2, x0)
 
   if (nargin >= 2 && nargin <= 7 && isvector (full (b)))
 
     if (ischar (A))
       A = str2func (A);
diff --git a/scripts/sparse/gmres.m b/scripts/sparse/gmres.m
--- a/scripts/sparse/gmres.m
+++ b/scripts/sparse/gmres.m
@@ -64,17 +64,17 @@
 ##
 ## @item @var{iter} is a vector containing the number of outer iterations and
 ## total iterations performed.
 ##
 ## @item @var{resvec} is a vector containing the relative residual at each
 ## iteration.
 ## @end itemize
 ##
-## @seealso{bicg, bicgstab, cgs, pcg}
+## @seealso{bicg, bicgstab, cgs, pcg, pcr, qmr}
 ## @end deftypefn
 
 function [x, flag, relres, it, resvec] = gmres (A, b, restart, rtol, maxit, M1, M2, x0)
 
   if (nargin < 2 || nargin > 8)
     print_usage ();
   endif
 
diff --git a/scripts/sparse/qmr.m b/scripts/sparse/qmr.m
--- a/scripts/sparse/qmr.m
+++ b/scripts/sparse/qmr.m
@@ -77,17 +77,17 @@
 ##
 ## @item
 ## R. Barrett, M. Berry, T. Chan, J. Demmel, J. Donato, J. Dongarra,
 ## V. Eijkhour, R. Pozo, C. Romine, and H. van der Vorst,
 ## @cite{Templates for the solution of linear systems: Building blocks
 ## for iterative methods}, SIAM, 2nd ed., 1994.
 ## @end enumerate
 ##
-## @seealso{bicg, bicgstab, cgs, gmres}
+## @seealso{bicg, bicgstab, cgs, gmres, pcg}
 ## @end deftypefn
 
 ## Author: Nathan Podlich <nathan.podlich@gmail.com>
 
 function [x, flag, relres, iter, resvec] = qmr (A, b, tol, maxit, M1, M2, x0)
 
   if (nargin >= 2 && isvector (full (b)))
 
